core/src/main/java/org/elasticsearch/search/SearchModule.java

```diff
@@ -310,15 +305,33 @@ public void registerFunctionScoreParser(ScoreFunctionParser<? extends ScoreFunct
      *        registered. So it is the name that the query should use as its {@link NamedWriteable#getWriteableName()}.
      */
     public <QB extends QueryBuilder<QB>> void registerQuery(Writeable.Reader<QB> reader, QueryParser<QB> parser, String... names) {
```

Comment:
that is a good point Christoph! if we really want to keep varargs otherwise we could validate that the array has length >=1 , I have no strong preference on how we solve it though